### PR TITLE
#143 track health on dogu cr

### LINKED ISSUE #143: Track dogu health on Dogu CRs

### PR DESCRIPTION
**Pull request by jelemux**
Resolves #143

### CODE REVIEW

**Review by alexander-dammeier — CHANGES_REQUESTED** [review]
Please check the kubebuilder version.

The rest looks good :)

File: api/v1/k8s.cloudogu.com_dogus.yaml [hunk 1]
```diff
@@ -3,7 +3,8 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.13.0
+    controller-gen.kubebuilder.io/version: v0.11.3
```
[the remark this answers is not in the document]
Comment (jelemux) [reply api/v1/k8s.cloudogu.com_dogus.yaml]: I had an older controller-gen installed locally. Upgraded and regenerated now.

File: controllers/health/availability.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package health
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+type AvailabilityChecker struct{}
```
[the remark this answers is not in the document]
Comment (jelemux) [reply controllers/health/availability.go:7]: I use it so I can encapsulate this complexity more easily and mock its behaviour. But maybe, mocking it is more trouble than simply using it directly as a function. Should I change this?

**Comment by alexander-dammeier** [comment]
the availability checker can stay this way. We can further refactor this in the future if it is too much abstraction.
The rest looks good to me.

**Pull request merged**